backend/src/main/java/com/example/backend/security/MongoUser.java

```diff
@@ -3,11 +3,15 @@
 import org.springframework.data.annotation.Id;
 import org.springframework.data.mongodb.core.mapping.Document;
 
+import java.util.List;
+
 @Document("users")
 public record MongoUser (
         @Id
         String id,
         String username,
-        String password
+        String password,
+        String email,
+        List<String> favorites
```

Comment:
you could make username or email or both unique, so that no one can register with the same email by using this annotation

`@Indexed(unique = true)`

Tutorial: https://www.baeldung.com/spring-data-mongodb-unique

just a suggestion :)